### PR TITLE
Record type

### PR DESCRIPTION
**Pull request by sondreso**
Introducing record types, also add tests for a couple of more scenarios

### CODE REVIEW

File: ert3/data/_record.py [hunk 1]
```diff
@@ -163,66 +187,69 @@ def __init__(self, name: str, storage_path: Path):
...
-            raise NotImplementedError(
-                f"cannot load record unless json, was {self._mime}"
-            )
-        async with aiofiles.open(str(self._uri)) as f:
-            contents = await f.read()
-            return Record(data=json.loads(contents))
+        if self._record_type != RecordType.LIST_BYTES:
+            async with aiofiles.open(str(self._uri)) as f:
+                contents = await f.read()
+                return Record(data=json.loads(contents))
```
Comment (jondequinor) [inline ert3/data/_record.py:245]: When did all records become json? Am confused

Comment (sondreso) [reply ert3/data/_record.py:245]: The records are not json, but the SharedStorageTransmitter chooses to store them as json?

File: ert3/data/_record.py [hunk 2]
```diff
@@ -40,6 +47,24 @@ def __init__(self, *, data, index=None, **kwargs):
...
+                return RecordType.LIST_BYTES
+        elif isinstance(self.data, Mapping):
+            if not self.data:
+                return RecordType.MAPPING_STR_FLOAT
+            if isinstance(list(self.data.keys())[0], int):
+                return RecordType.MAPPING_INT_FLOAT
+            if isinstance(list(self.data.keys())[0], str):
+                return RecordType.MAPPING_STR_FLOAT
+        raise ValueError("Not able to deduce record type from data")
+
```
Comment (jondequinor) [inline ert3/data/_record.py:67]: TypeError maybe. This value does not have the expected type.

**Review by jondequinor — APPROVED** [review]
Good stuff

**Pull request merged**